WPFHospitalEditor/AppointmentAnalysisWindow.xaml.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
+using WPFHospitalEditor.Controller;
+using WPFHospitalEditor.Controller.Interface;
+using WPFHospitalEditor.DTOs;
+using WPFHospitalEditor.Model;
+
+namespace WPFHospitalEditor
+{
+    /// <summary>
+    /// Interaction logic for AppointmentAnalysisWindow.xaml
+    /// </summary>
+    public partial class AppointmentAnalysisWindow : Window
+    {
+        private readonly IExaminationServerController examinationServerController = new ExaminationServerController();
+        private readonly ISchedulingServerController schedulingController = new SchedulingServerController();
+        public ObservableCollection<ExaminationWithAvailableRescheduling> examinationWithAvailableRescheduling { get; set; }
+        public AppointmentAnalysisWindow(int specialtyId)
+        {
+            InitializeComponent();
+            FillObservableCollection(specialtyId);
+            this.DataContext = this;
+        }
+
+        private void FillObservableCollection(int specialtyId)
+        {
+            examinationWithAvailableRescheduling = new ObservableCollection<ExaminationWithAvailableRescheduling>();
+            List<Examination> examinations = examinationServerController.GetBySpecialtyId(specialtyId).ToList();
+            foreach (Examination exam in examinations) {
+                ExaminationWithAvailableRescheduling examination = new ExaminationWithAvailableRescheduling(exam.Id,exam.PatientId, exam.DoctorId, exam.Priority, exam.TimeInterval.Start, DateTime.Now, exam.RequiredSpecialtyId);
+                examinationWithAvailableRescheduling.Add(examination);
+                examinationSearchComboBox.Items.Add(exam.Id);
+            }
```

Comment:
You should filter List in SQL query, because in this case you get all the data, and that's a lot of data, and then you iterate through that data and filter. That may be fine for now, but in future it will be problematic because it will be too slow.

Reply:
I'm not filtering anything here, just attaching available appointments for rescheduling for every examination, and user chooses which one he wants to reschedule

Reply:
My mistake